src/timepicker/timepicker.js

```diff
@@ -194,23 +202,27 @@ class TimePicker extends React.Component<TimePickerPropsT, TimePickerStateT> {
     excludeOptions,
     newProps,
   ) => {
-    const result = stringToOptions(filterValue, this.props.format);
+    const result = this.stringToOptions(filterValue, this.props.format);
     if (result.length) {
       return result;
     }
     return filterOptions(options, filterValue, excludeOptions, newProps);
   };
 
-  buildSelectedOption(value: Date, format: '12' | '24' = '12') {
-    const secs = dateToSeconds(value);
+  buildSelectedOption: (T, ?'12' | '24') => OptionT = (
+    value,
+    format = '12',
+  ) => {
```

Comment:
flow is thankfully working on inferring generics from outside of the function scope (skolemization) https://medium.com/flow-type/what-were-building-in-2020-bcb92f620c75 until then, we need to provide an 'input' to the function like below

```suggestion
  buildSelectedOption = <T>(
    value: T,
    format: '12' | '24' = '12',
  ): OptionT => {
```

Reply:
We decided to punt on this change, because it would require manually passing the type all the call site like `buildSelectedOption<Date>()`